app/common/model/src/main/java/io/syndesis/common/model/connection/PropertyRelation.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.common.model.connection;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+@Value.Immutable
+@JsonDeserialize(builder = PropertyRelation.Builder.class)
+@SuppressWarnings("immutables")
+public interface PropertyRelation extends Serializable {
+
+    class Builder extends ImmutablePropertyRelation.Builder {
+        // make ImmutablePropertyRelation.Builder accessible
+    }
+
+    @Value.Immutable
+    @JsonDeserialize(builder = PropertyRelationWhen.Builder.class)
+    interface PropertyRelationWhen {
+
+        @SuppressWarnings("PMD.UseUtilityClass")
+        class Builder extends ImmutablePropertyRelationWhen.Builder {
+        }
+
+        String getId();
+
+        String getValue();
+    }
+
+    String getAction();
+
+    List<PropertyRelationWhen> getWhen();
```

Comment:
Should it be then `WhenList` ?

Really not sure about the naming. At least its not very intuitive.

Reply:
It's the way UI asked for it to be. Field names make more sense when viewing an instance data model though.